packages/components/src/components/text/text.jsx

```diff
@@ -2,14 +2,13 @@ import classNames from 'classnames';
 import React from 'react';
 import PropTypes from 'prop-types';
 
-const Text = ({ children, size, color, align, weight, line_height, as, className, ...props }) => {
-    const style = {
+const Text = ({ children, size, color, align, weight, line_height, as, className, styles, ...props }) => {
+    const style = styles || {
```

Comment:
We should only have the styles `prop` as an additional way to deal with edge cases so we should keep the class styles as well.
Example:

- e.g - `const styles = { fontWeight: 'bold' , ....other_styles };`
`// styles should be an object with css applied style properties `

I suggest an approach like something below:

```
const Text = ({...., styles}) => {
const setStyles = () => { 
const class_styles = { '---text-size: 'var(--text-size-${size || 's'}', ... };
if (!isEmptyObject(styles)) {
// if styles prop is set, merge both class styles and styles from props
return { ...class_styles, ...styles };
}
return class_styles;
}
...
(as === 'p' && (
<p {...props} style={setStyles()}>{children}</p>
        )) || (          
 <span {...props} className={class_names} style={setStyle()}>{children}</span>
))};
```